sources/components/markdown/MermaidRenderer.tsx

```
@@ -0,0 +1,218 @@
+    const { theme } = useUnistyles();
+    const [dimensions, setDimensions] = React.useState({ width: 0, height: 200 });
+    const [svgContent, setSvgContent] = React.useState<string | null>(null);
+
+    const onLayout = React.useCallback((event: any) => {
+        const { width } = event.nativeEvent.layout;
+        setDimensions(prev => ({ ...prev, width }));
+    }, []);
+
+    // Web platform uses direct SVG rendering for better performance and native DOM integration
+    if (Platform.OS === 'web') {
+        const [hasError, setHasError] = React.useState(false);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)